### PR TITLE
smartlink to fallback on en-US

### LINKED ISSUE #3053: smartLink() should fall back to English in some way
We have a lot of xrefs and sidebars (soon!) that use `smartLink()` and these can recover to fall back to English if the translated content doesn't exist. 

We can discuss implementation choices here. For example:

1. `<a class="page-not-created" title="Not yet created ...">SomeApiMethod</a> <a href="/en-US/docs/SomeApiMethod">English</a>`
2. `<a title="Not available in Japanese. Falling back to English" href="/en-US/docs/SomeApiMethod">SomeApiMethod</a>`
3. `<small>In English</small> <a href="/en-US/docs/SomeApiMethod">SomeApiMethod</a>`
4. (Chris's suggestion) `<a title="Page only currently available in English" href="/en-US/docs/SomeApiMethod">SomeApiMethod (en-US)</a>`

Any other ideas?

### PR DESCRIPTION
**Pull request by peterbe**
Fixes #3053

Before (http://localhost:3000/ca/docs/Web/JavaScript/Reference/Global_Objects/Array/reverse):
Screen Shot 2021-03-04 at 12 25 56 PM

After:
Screen Shot 2021-03-04 at 12 25 35 PM

Before (http://localhost:3000/de/docs/Web/CSS/Reference):
Screen Shot 2021-03-04 at 12 25 25 PM

After:
Screen Shot 2021-03-04 at 12 25 03 PM

### CODE REVIEW

**Comment by peterbe** [comment]
@chrisdavidmills  I basically just took [your suggestion](https://github.com/mdn/yari/issues/3053#issuecomment-789611237). 

@schalkneethling Please review my use of the `<small>` tag for the `(en-US)` portion. 

@fiji-flo Would it be better to go through the `translationsOf` Map object?

**Review by fiji-flo — APPROVED** [review]
❤️

File: kumascript/src/api/web.js [hunk 1]
```diff
@@ -83,6 +83,27 @@ module.exports = {
...
+      // change it to the en-US URL instead.
+      const hrefSplit = href.split("/");
+      hrefSplit[1] = "en-US";
+      const enUSPage = this.info.getPageByURL(hrefSplit.join("/"));
+      if (enUSPage.url) {
+        // But it's still a flaw. Record it so that translators can write a
+        // translated document to "fill the hole".
+        flaw = this.env.recordNonFatalError(
+          "broken-link",
+          `${hrefpath} does not exist but fallbacked on ${enUSPage.url}`
```
Comment (escattone) [inline kumascript/src/api/web.js:97]:
```suggestion
          `${hrefpath} does not exist, fallback to ${enUSPage.url}`
```

**Review by escattone — COMMENTED** [review]
On my phone, so unable to test. Just a tiny nit for now.

**Review by escattone — APPROVED** [review]
Love it, looks and works great, thanks @peterbe!

**Comment by escattone** [comment]
@schalkneethling I know you have a lot on your plate, but would you mind giving this a quick review for any a11y concerns, specifically regarding the use of `<small>` tag?

File: kumascript/src/api/web.js [hunk 2]
```diff
@@ -83,6 +83,27 @@ module.exports = {
...
+          "broken-link",
+          `${hrefpath} does not exist, fallback to ${enUSPage.url}`
+        );
+        flawAttribute = ` data-flaw-src="${util.htmlEscape(flaw.macroSource)}"`;
+        return (
+          '<a class="page-only-in-english" ' +
+          'title="Page currently only available in English" ' +
+          `href="${enUSPage.url}"${flawAttribute}>${content} <small>(en-US)</small></a>`
+        );
+      }
```
Comment (schalkneethling) [inline kumascript/src/api/web.js:105]:
```suggestion
        return (
          '<a class="page-only-in-english" ' +
          'title="Page currently only available in English" ' +
          `href="${enUSPage.url}"${flawAttribute}>${content} <span>(en-US)</span></a>`
        );
```

Then in `client/src/document/index.scss` add the following:

```
.page-only-in-english {
  span {
    font-size: $small-font-size;
  }
}
```

**Review by schalkneethling — CHANGES_REQUESTED** [review]
This is going to be so awesome. Thank you, Peter. Just one change recommendation.

File: kumascript/src/api/web.js (hunk 2 above)
Comment (schalkneethling) [reply kumascript/src/api/web.js]: I know, sometimes I want to change my mind about the use of these elements but, we would need CSS anyhow to ensure that the `small` element uses the `$small-font-size`. I am kinda undecided on that one.

**Review by schalkneethling — APPROVED** [review]
🎉  - Thank you @peterbe

**Pull request merged**